src/main/java/com/ciandt/techgallery/persistence/dao/storage/StorageDAOImpl.java

```diff
@@ -0,0 +1,177 @@
+package com.ciandt.techgallery.persistence.dao.storage;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.InputStreamContent;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.StorageScopes;
+import com.google.api.services.storage.model.Bucket;
+import com.google.api.services.storage.model.BucketAccessControl;
+import com.google.api.services.storage.model.ObjectAccessControl;
+import com.google.api.services.storage.model.StorageObject;
+
+import com.ciandt.techgallery.persistence.dao.StorageDAO;
+import com.ciandt.techgallery.utils.TechGalleryUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+
+/**
+ * Class that implements the StorageDAO.
+ *
+ * @author <a href="mailto:joaom@ciandt.com"> João Felipe de Medeiros Moreira </a>
+ * @since 13/10/2015
+ *
+ */
+public class StorageDAOImpl implements StorageDAO {
+
+  /*
+   * Constants --------------------------------------------
+   */
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String APPLICATION_NAME = "[[tech-gallery]]";
```

Comment:
@jfmedeiros1820 please use the application ID instead of a hard-coded string as per https://cloud.google.com/appengine/docs/java/javadoc/com/google/appengine/api/utils/SystemProperty